### DISCUSSION TITLE
How to use a version from a pull request?

### DISCUSSION THREAD

**Discussion by rchavik** [storybookjs/react-native#193] [Q&A] [answered]
I tried to use https://github.com/storybookjs/react-native/pull/190 by `npm install storybookjs/react-native#pull/190/head` but getting the following error: `Module not found: Can't resolve '@storybook/react-native' in '<$srcroot>/storybook'` when opening from the web and android (haven't gotten into ios yet).

Is this a supported workflow or I need to wait until the proper fix is in?

**Comment by dannyhw** [comment, marked as answer]
I usually use the github cli which makes it easy to get the pr checked out `gh pr checkout 190`.

The issue with a module not found seems more like an issue with your setup. Have you ran bootstrap? Also in the latest version of the repo you need to manually run the metro bundler (yarn start) in the example/native folder since the auto launched one doesn't work.

**Reply by rchavik** [reply]
> I usually use the github cli which makes it easy to get the pr checked out `gh pr checkout 190`.
>
> The issue with a module not found seems more like an issue with your setup. Have you ran bootstrap? Also in the latest version of the repo you need to manually run the metro bundler (yarn start) in the example/native folder since the auto launched one doesn't work.

Sorry I wasn't clear. That sounds like the step if I want to view/use the PR in my local git clone.
 
My aim was to use the PR so that it gets installed in my project's `node_modules`. 

I had a test project running using the following:
- a test project created with `expo`, 
- ran `sb init`.
- Made the changes to `App.tsx` and `storybook` directory,
- `expo start -c`, then use `a` to run it in android simulator.  It is working, I can browse the Button stories.
- Tried running the web version via `expo` cli using `w` key.
- Browser opens with the `TypeError`, can't remember the exact error, but it was related to prop-types or RNViewProps.

So this is why I wanted to try if using the changes in PR solves it.

**Reply by dannyhw** [reply]
How did you add it to the project? Did you add the git branch as a dependency in your node_modules?

**Reply by rchavik** [reply]
Using `npm install storybookjs/react-native#pull/190/head` (taken from https://stackoverflow.com/a/36743127)

**Reply by dannyhw** [reply]
Ok well I think this won't work because the project is not released in this way. This repo is a mono repo so you would need to checkout the app/react-native folder and not the entire repository. Even if you can do this you need to run the prepare command to provide the dist/ folder which is where the bundled code is located.

In summary It's probably not viable to do it this way. 

You could perhaps do something a bit more custom by checking out the code and running the prepare then you would copy the folder into your project.

However you should also bear in mind that this code on this PR is part of the unfinished and unreleased 6.0 alpha.

**Reply by rchavik** [reply]
Thanks for the explanation!
